### PR TITLE
testing tests

### PR DESCRIPTION
**Pull request by jywarren**
(ignore, just confirming tests are running!)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ env:
     - COMPOSE_FILE=./containers/docker-compose-testing.yml
   matrix:
     - TASK="rake test:all"
+    - TASK="rails test"
```
Comment (jywarren) [inline .travis.yml]: I've added this line to see if it gets regular (unit, functional, integration) tests running again?

**Comment by jywarren** [comment]
Ok! I think that did it! @alaxalves take a look 

https://travis-ci.org/publiclab/plots2/jobs/544975404

**Comment by jywarren** [comment]
Errors:

```
ActionView::Template::Error:         ActionView::Template::Error: undefined method `path' for "awesome":String
        Did you mean?  pathmap
            app/views/search/tags.html.erb:12:in `block in _app_views_search_tags_html_erb___2889796601069456699_47004343835500'
```

```
ActionView::Template::Error:         ActionView::Template::Error: undefined method `count' for nil:NilClass
        Did you mean?  concat
            app/views/stats/_range.html.erb:186:in `_app_views_stats__range_html_erb__1237107643757034086_47004338106100'
```

```
NoMethodError:         NoMethodError: undefined method `nid' for "blog":String
        Did you mean?  nil?
            app/api/srch/search.rb:163:in `block (3 levels) in <class:Search>'
```

```
NoMethodError:         NoMethodError: undefined method `answer_comment_notify' for #<Comment:0x000055801675dbb8>
            app/models/comment.rb:237:in `add_answer_comment'
```

```
 test_should_save_comment#CommentTest (53.47s)
        Expected nil to be truthy.
        test/unit/comment_test.rb:13:in `block in <class:CommentTest>'
```

```
ActionView::Template::Error:         ActionView::Template::Error: undefined method `cid' for nil:NilClass
            app/views/notes/_comments.html.erb:8:in `block in _app_views_notes__comments_html_erb__94806921206799871_47004356406460'
```

I think that's it. It's not so bad. Let's fix these and merge this.

File: app/views/stats/_range.html.erb [hunk 2]
```diff
@@ -183,7 +183,7 @@ master
 </div>
 <script src="https://cdn.jsdelivr.net/npm/apexcharts"></script>
 <script>
-var questions = [<%= @questions.count %>];
+var questions = [<%= @questions %>];
```
Comment (jywarren) [inline app/views/stats/_range.html.erb]: Hi @cesswairimu can you check my work here? These tests had been silently failing, they're not too bad, just trying to resolve them all. Thanks!

**Comment by jywarren** [comment]
OK, so we eventually want to run all tests EXCEPT system tests. And we can break out functional, unit, integration into parallel processes.

**Comment by jywarren** [comment]
OK, another error here:

https://github.com/publiclab/plots2/blob/ef2e639cfc58b1b7cb5cf246161d56591c9a3537/app/views/stats/index.html.erb#L187

values is on a nil object...

**Comment by jywarren** [comment]
* [ ] `@graph_notes` is nil
* [ ] Also:

```
 test_running_search_tags#SearchServiceTest (149.60s)
        Expected: 1
          Actual: 4
        test/unit/api/search_service_test.rb:64:in `block in <class:SearchServiceTest>'
```

**Comment by jywarren** [comment]
Unit tests first:

```

 FAIL["test_running_search_tags", #<Minitest::Reporters::Suite:0x00005582525dac30 @name="SearchServiceTest">, 20.01505934300002]
 test_running_search_tags#SearchServiceTest (20.02s)
        Expected: 1
          Actual: 4
        test/unit/api/search_service_test.rb:64:in `block in <class:SearchServiceTest>'
```

```
FAIL["test_should_parse_incoming_mail_from_other_domain_who_use_gmail_service_correctly_and_add_answer_comment", #<Minitest::Reporters::Suite:0x000055825b8e5bd8 @name="CommentTest">, 21.882848784000032]
 test_should_parse_incoming_mail_from_other_domain_who_use_gmail_service_correctly_and_add_answer_comment#CommentTest (21.88s)
        --- expected
        +++ actual
        @@ -1 +1,12 @@
        -"Answer comment two"
        +"This is another reply by email comment
        +  
        +<!-- @@$$%% Trimmed Content @@$$%% -->On Tue, Jul 3, 2018 at 11:17 PM Naman Gupta \\<[01namangupta@gmail.com](mailto:01namangupta@gmail.com)\\> wrote:  
        +
        +> This is reply by comment
        +>   
        +> 
        +> On Tue, Jul 3, 2018 at 11:13 PM Rails Projects \\<[railsprojects2018@gmail.com](mailto:railsprojects2018@gmail.com)\\> wrote:  
        +> 
        +> > This is a comment sent to the user by publiclab.
        +
        +"
        test/unit/comment_test.rb:304:in `block in <class:CommentTest>'
```

**Comment by jywarren** [comment]
Then functional tests:

```

ERROR["test_search_tags_at_/search/tags/awesome", #<Minitest::Reporters::Suite:0x000055937f175c88 @name="SearchControllerTest">, 31.13762619199997]
 test_search_tags_at_/search/tags/awesome#SearchControllerTest (31.14s)
NoMethodError:         NoMethodError: undefined method `nid' for "awesome":String
        Did you mean?  nil?
            test/functional/search_controller_test.rb:57:in `block in <class:SearchControllerTest>'
ERROR["test_should_decrease_cached_likes_if_unliked", #<Minitest::Reporters::Suite:0x0000559387094f50 @name="AnswerLikeControllerTest">, 39.90560243499999]
 test_should_decrease_cached_likes_if_unliked#AnswerLikeControllerTest (39.91s)
ActionController::UrlGenerationError:         ActionController::UrlGenerationError: No route matches {:action=>"likes", :aid=>1, :controller=>"answer_like"}
            test/functional/answer_like_controller_test.rb:35:in `block (2 levels) in <class:AnswerLikeControllerTest>'
            test/functional/answer_like_controller_test.rb:34:in `block in <class:AnswerLikeControllerTest>'
ERROR["test_should_get_likes", #<Minitest::Reporters::Suite:0x000055938706ed00 @name="AnswerLikeControllerTest">, 39.92457790799995]
 test_should_get_likes#AnswerLikeControllerTest (39.92s)
ActionController::UrlGenerationError:         ActionController::UrlGenerationError: No route matches {:action=>"likes", :aid=>1, :controller=>"answer_like"}
            test/functional/answer_like_controller_test.rb:17:in `block in <class:AnswerLikeControllerTest>'
ERROR["test_should_get_show", #<Minitest::Reporters::Suite:0x0000559387062668 @name="AnswerLikeControllerTest">, 39.931491367000035]
 test_should_get_show#AnswerLikeControllerTest (39.93s)
ActionController::UrlGenerationError:         ActionController::UrlGenerationError: No route matches {:action=>"show", :controller=>"answer_like", :id=>1}
            test/functional/answer_like_controller_test.rb:10:in `block in <class:AnswerLikeControllerTest>'
ERROR["test_should_increase_cached_likes_if_liked", #<Minitest::Reporters::Suite:0x00005593870442a8 @name="AnswerLikeControllerTest">, 39.94734042999994]
 test_should_increase_cached_likes_if_liked#AnswerLikeControllerTest (39.95s)
ActionController::UrlGenerationError:         ActionController::UrlGenerationError: No route matches {:action=>"likes", :aid=>1, :controller=>"answer_like"}
            test/functional/answer_like_controller_test.rb:26:in `block (2 levels) in <class:AnswerLikeControllerTest>'
            test/functional/answer_like_controller_test.rb:25:in `block in <class:AnswerLikeControllerTest>'

```

Then integration tests:

```
 FAIL["test_should_parse_incoming_mail_from_yahoo_service_correctly_and_add_answer_comment", #<Minitest::Reporters::Suite:0x000055e6700c7608 @name="YahooParsingTest">, 4.274443044999998]
 test_should_parse_incoming_mail_from_yahoo_service_correctly_and_add_answer_comment#YahooParsingTest (4.27s)
        --- expected
        +++ actual
        @@ -1 +1,11 @@
        -"Answer comment two"
        +"This is reply by mail comment from yahoooo
        +
        +<!-- @@$$%% Trimmed Content @@$$%% --> On Tuesday, 3 July 2018, 11:20:57 PM IST, Rails Projects \\<railsprojects2018@gmail.com\\> wrote: 
        +
        +  
        +
        +  
        +
        +This is a comment sent to the user by publiclab.  
        +
        +"
        test/integration/incoming_mail_parsing_test/yahoo_parsing_test.rb:14:in `block in <class:YahooParsingTest>'
 FAIL["test_should_parse_incoming_mail_from_outlook_service_correctly_and_add_answer_comment", #<Minitest::Reporters::Suite:0x000055e66e36c360 @name="OutlookParsingTest">, 12.067988039]
 test_should_parse_incoming_mail_from_outlook_service_correctly_and_add_answer_comment#OutlookParsingTest (12.07s)
        --- expected
        +++ actual
        @@ -1 +1,30 @@
        -"Answer comment two"
        +"\r This is the second reply sent by the user.
        +<!-- @@$$%% Trimmed Content @@$$%% -->* * *
        +
        +<font face=\"Calibri, sans-serif\" style=\"font-size:11pt\" color=\"#000000\"><b>From:</b> TestUser PublicLab<br>\r
        +<b>Sent:</b> Wednesday, December 12, 2018 8:43 PM<br>\r
        +<b>To:</b> kevinzluo PublicLab<br>\r
        +<b>Subject:</b> Re: (#a2)</font>
        +&nbsp;
        +
        +<style type=\"text/css\" style=\"display:none\">\r
        +<!--\r
        +p\r
        +\t{margin-top:0;\r
        +\tmargin-bottom:0}\r
        +-->\r
        +</style>
        +
        +\r This is the first reply sent by the user.
        +
        +* * *
        +
        +<font face=\"Calibri, sans-serif\" color=\"#000000\" style=\"font-size:11pt\"><b>From:</b> kevinzluo PublicLab &lt;pl.kevinzluo@gmail.com&gt;<br>\r
        +<b>Sent:</b> Wednesday, December 12, 2018 8:42 PM<br>\r
        +<b>To:</b> pl.usertest@outlook.com<br>\r
        +<b>Subject:</b> (#a2)</font>
        +&nbsp;
        +
        +This is the email sent to the user by PublicLab.
        +
        +"
        test/integration/incoming_mail_parsing_test/outlook_parsing_test.rb:12:in `block in <class:OutlookParsingTest>'
 FAIL["test_should_parse_incoming_mail_from_gmail_service_correctly_and_add_answer_comment", #<Minitest::Reporters::Suite:0x000055e675a5f058 @name="GmailParsingTest">, 26.546761369000023]
 test_should_parse_incoming_mail_from_gmail_service_correctly_and_add_answer_comment#GmailParsingTest (26.55s)
        --- expected
        +++ actual
        @@ -1 +1,12 @@
        -"Answer comment two"
        +"This is another reply by email comment
        +  
        +<!-- @@$$%% Trimmed Content @@$$%% -->On Tue, Jul 3, 2018 at 11:17 PM Naman Gupta \\<[01namangupta@gmail.com](mailto:01namangupta@gmail.com)\\> wrote:  
        +
        +> This is reply by comment
        +>   
        +> 
        +> On Tue, Jul 3, 2018 at 11:13 PM Rails Projects \\<[railsprojects2018@gmail.com](mailto:railsprojects2018@gmail.com)\\> wrote:  
        +> 
        +> > This is a comment sent to the user by publiclab.
        +
        +"
        test/integration/incoming_mail_parsing_test/gmail_parsing_test.rb:12:in `block in <class:GmailParsingTest>'
```

**Comment by jywarren** [comment]
OK, again - functionals:

```

ERROR["test_should_assign_correct_value_to_graph_notes_on_GET_stats", #<Minitest::Reporters::Suite:0x000055c256e9b808 @name="StatsControllerTest">, 14.618104431000006]
 test_should_assign_correct_value_to_graph_notes_on_GET_stats#StatsControllerTest (14.62s)
ActionView::Template::Error:         ActionView::Template::Error: undefined method `values' for nil:NilClass
            app/views/stats/index.html.erb:187:in `_app_views_stats_index_html_erb__2863561479807605957_47146587739240'
            test/functional/stats_controller_test.rb:14:in `block in <class:StatsControllerTest>'


ERROR["test_should_assign_correct_value_to_graph_wikis_on_GET_stats", #<Minitest::Reporters::Suite:0x000055c256ce1bc0 @name="StatsControllerTest">, 14.656739618000017]
 test_should_assign_correct_value_to_graph_wikis_on_GET_stats#StatsControllerTest (14.66s)
ActionView::Template::Error:         ActionView::Template::Error: undefined method `values' for nil:NilClass
            app/views/stats/index.html.erb:187:in `_app_views_stats_index_html_erb__2863561479807605957_47146587739240'
            test/functional/stats_controller_test.rb:22:in `block in <class:StatsControllerTest>'


ERROR["test_search_tags_at_/search/tags/awesome", #<Minitest::Reporters::Suite:0x000055c2606ef1e0 @name="SearchControllerTest">, 96.23664567800006]
 test_search_tags_at_/search/tags/awesome#SearchControllerTest (96.24s)
NoMethodError:         NoMethodError: undefined method `nid' for "awesome":String
        Did you mean?  nil?
            test/functional/search_controller_test.rb:57:in `block in <class:SearchControllerTest>'
```

**Comment by jywarren** [comment]
The integration tests are failing because they test responses to answers, not comments. @ViditChitkara if I can't complete this today, i'd appreciate any help (perhaps a PR against this one?) refactoring those tests for comments. They're a bit complex: https://github.com/publiclab/plots2/tree/testing-tests/test/integration/incoming_mail_parsing_test 

These tests have been failing but recent changes to our test runner was causing them not to appear! Doing our best to patch this!

**Comment by jywarren** [comment]
We need to refactor this one:

https://github.com/publiclab/plots2/blob/ca3f504ce17180c2aa864db03e625df21c7c5754/test/fixtures/incoming_test_emails/outlook/incoming_outlook_email.eml#L130

To be for a comment like this one:

https://github.com/publiclab/plots2/blob/ca3f504ce17180c2aa864db03e625df21c7c5754/test/fixtures/incoming_test_emails/yahoo/incoming_yahoo_email.eml#L49

**Comment by jywarren** [comment]
On this error: 
```
ActionView::Template::Error: undefined method `values' for nil:NilClass
            app/views/stats/index.html.erb:187:in `_app_views_stats_index_html_erb
```

I'm not sure what's going on because this doesn't seem like it should be erroring:

https://github.com/publiclab/plots2/pull/4887/files#diff-834bd3009431ee7aa94033f356f5a2ceR58

**Comment by jywarren** [comment]
OK, it was an issue of `.to_s` and `.to_time` in stats controller.

**Comment by ViditChitkara** [comment]
I think they are still failing will have a look.

**Comment by jywarren** [comment]
Some progress. Actually i had to do a partial fix for #5916 -- it doesn't thread them but it does detect the `(#c0000)` style email subject lines, it just treats them as regular comments, not threaded ones.

**Comment by jywarren** [comment]
OK, the only errors left are 3 of the same:

```
ActionView::Template::Error: undefined method `values' for nil:NilClass
            app/views/stats/index.html.erb:187:in `_app_views_stats_index_html_erb
```

Happening on these lines:

```
            test/integration/I18n_test.rb:526
            test/functional/stats_controller_test.rb:14
            test/functional/stats_controller_test.rb:22
```

**Comment by jywarren** [comment]
@cesswairimu do you think you could take a look here? I really thought I fixed this in 4df3232031ae14dfdd97dca057c1816851494a12 but I'm still seeing it happen 3 times. Any ideas? I'm pretty stuck 🤷‍♂

**Comment by ViditChitkara** [comment]
I think the stats error is fixed. Just the email test faliure left. @jywarren would you mind taking a look at the changes I made in this pr?
I'll start with #5916 to fix the email error, right?

File: app/views/stats/_range.html.erb (hunk 2 above)
Comment (cesswairimu) [reply app/views/stats/_range.html.erb]: Sorry I missed this, Taking a look now,

**Comment by jywarren** [comment]
On the email error I thought we could just create a comment, but not a
reply comment, for the short term fix. So I think it should be an easy fix
but I wasn't sure why it was still failing. Thanks for the help!!!

**Comment by cesswairimu** [comment]
@jywarren @ViditChitkara I inspected the values we were using and the `reply_to` was nil and I think this makes sense from your description on #5916 

email

that we removed the reply to for using emails to comment(let me know if I got this wrong). So I though we could remove this assertion until the feature is re-implemented.

**Comment by jywarren** [comment]
Ok awesome this seems right until #5916 is done. Thanks Cess! Thanks Vidit! I hope this passes!!!

**Comment by jywarren** [comment]
Awesome work!!!!!! Thanks for the help!!!!!

**Comment by ViditChitkara** [comment]
Great!! Moving on to #5916 now.

**Pull request merged**